### PR TITLE
vile 9.8v (new formula)

### PR DESCRIPTION
**Pull request by johnsonjh**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Add vile 9.8v 2021-12-12

### CODE REVIEW

File: Formula/vile.rb
Comment (SMillerDev) [inline Formula/vile.rb:15]:
```suggestion
    system "./configure", *std_configure_args,
```

File: Formula/vile.rb
Comment (SMillerDev) [inline Formula/vile.rb:24]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

File: Formula/vile.rb
Comment (SMillerDev) [inline Formula/vile.rb:8]: Why not use the one on macOS?

File: Formula/vile.rb
Comment (johnsonjh) [reply Formula/vile.rb:24]: I'll try to make it process a file and check the output. I'll update it today.

File: Formula/vile.rb
Comment (johnsonjh) [reply Formula/vile.rb]: Fixed in https://github.com/Homebrew/homebrew-core/pull/106086/commits/66ecf45918a2cedd7ace3ef6ed34a785203da86a.

File: Formula/vile.rb
Comment (johnsonjh) [reply Formula/vile.rb]: Fixed in https://github.com/Homebrew/homebrew-core/pull/106086/commits/0b036532ece5374dfa7d0b915f54b87c03f828b4.

File: Formula/vile.rb
Comment (johnsonjh) [reply Formula/vile.rb:30]: Fixed in https://github.com/Homebrew/homebrew-core/pull/106086/commits/89a13002c1bd3cde72283a672f667c23fa608b37#diff-3fb6d8265767b629e5c8862db30f7a88d664cd63a86ec9ebfae31ff209063b83R31 - now it uses expect to start the editor, create a new file, quit cleanly, and ensure the file was created.

File: Formula/vile.rb
Comment (carlocab) [inline Formula/vile.rb:28]: Can we use Ruby's `PTY` module instead of `expect`?

File: Formula/vile.rb
Comment (carlocab) [inline Formula/vile.rb:8]: Or, if we need to use `expect`, we should make this `uses_from_macos`.

File: Formula/vile.rb
Comment (johnsonjh) [reply Formula/vile.rb:28]: I'm not sure...  I've never touched Ruby before 48 hours ago!  (I'm a C and assembly person, so I'll defer to the judgement of the committers here.)

Comment (carlocab) [reply Formula/vile.rb:28]: We probably can -- `grep -r 'require "pty"` for many examples.

File: Formula/vile.rb
Comment (johnsonjh) [reply Formula/vile.rb:8]: There are some "tests" that can be run without expect, but they'd be limited to something like a simple version check, or using one of the included helper tools like `vile-to-html`, but I think the current test is "better".

**Comment by johnsonjh** [comment]
@carlocab I'll wait for you guidance here ... if `expect` is OK to use I'll make the change suggested above.

File: Formula/vile.rb
Comment (carlocab) [inline Formula/vile.rb:28]: Something like this should work.
```suggestion
    require "pty"

    PTY.spawn(bin/"vile") do |_, w,|
      w.write ":w new\n"
      w.write ":q\n"
    end
```

File: Formula/vile.rb
Comment (carlocab) [inline Formula/vile.rb:4]: Is `v` part of the version? `brew` probably won't pick that up; see the output of `brew info vile` to check. If it doesn't:
```suggestion
  url "https://invisible-island.net/archives/vile/current/vile-9.8v.tgz"
  version "9.8v"
```

Comment (johnsonjh) [reply Formula/vile.rb:4]:
```sh
brew info vile
vile: stable 9.8v
Vi Like Emacs Editor
https://invisible-island.net/vile/
/usr/local/Cellar/vile/9.8v (221 files, 7.8MB) *
```

It seem OK as is :)

File: Formula/vile.rb
Comment (johnsonjh) [reply Formula/vile.rb:12]: I think it should be OK without out it, I'll have to see what happens with CI on linux.

File: Formula/vile.rb
Comment (carlocab) [inline Formula/vile.rb:32]: Yea, ok, this didn't work. I guess we can use `expect` 🙂

Comment (johnsonjh) [reply Formula/vile.rb]: I'll change it back to expect - :)

Comment (johnsonjh) [reply Formula/vile.rb]: https://github.com/Homebrew/homebrew-core/pull/106086/commits/701d9ee62b34b7b60994866fc7fe723025359f47 ...

File: Formula/vile.rb
Comment (carlocab) [inline Formula/vile.rb:26]: Let's try to make this easier to read. The `expect` script may need tweaking. We need to do `\\r` to avoid `ruby` expanding the `\r`. (Or maybe it works without it, not sure.)
```suggestion
    ENV["TERM"] = "xterm"
    (testpath/"vile.exp").write <<~EOS
      spawn #{bin}/vile
      expect "unnamed"
      send ":w new\\r:q\\r"
      expect eof
    EOS
    system "expect", "-f", "vile.exp"
```

Comment (johnsonjh) [reply Formula/vile.rb:26]: Let me try something like that.  And see why Ubuntu is now failing ... hopefully about done.

Comment (johnsonjh) [reply Formula/vile.rb]: Should be fixed in https://github.com/Homebrew/homebrew-core/pull/106086/commits/5a52954a6f9b43068761129475d121390bb23d20

File: Formula/vile.rb
Comment (johnsonjh) [reply Formula/vile.rb]: Should be good to go in the latest commit.  Thanks for the xcrypt hint.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**
